### PR TITLE
Add XaoS

### PR DESCRIPTION
**Pull request by Mailaender**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/xaos-project/XaoS/pull/233
- [X] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)* https://github.com/xaos-project/XaoS/pull/233

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by Mailaender** [comment]
bot, build io.github.xaos_project.XaoS

File: xaos.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+       fractal types is provided.
+       Other features include autopilot mode, palette changing,
+       image saving, fractal inversion, filters, and a built in fractal
+       tutorial.
+    </p>
+  </description>
+
+  <content_rating type="oars-1.1" />
+
+  <launchable type="desktop-id">io.github.xaos-project.desktop</launchable>
```
Comment (hfiguiere) [inline xaos.appdata.xml:26]:
```suggestion
  <launchable type="desktop-id">io.github.xaos_project.XaoS.desktop</launchable>
```

File: xaos.appdata.xml [hunk 2]
```diff
@@ -0,0 +1,74 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+  <id>io.github.xaos_project.XaoS</id>
```
Comment (hfiguiere) [inline xaos.appdata.xml:3]: but why is this here since you have a patch for upstream's?

File: appdata3.patch [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
++  <id>io.github.xaos_project.XaoS</id>
+   <metadata_license>FSFAP</metadata_license>
+   <project_license>GPL-2.0+</project_license>
+   <name>XaoS</name>
+@@ -21,7 +21,7 @@
+     </p>
+   </description>
+ 
+-  <launchable type="desktop-id">io.github.xaos-project.desktop</launchable>
++  <launchable type="desktop-id">io.github.xaos_project.desktop</launchable>
```
Comment (hfiguiere) [inline appdata3.patch:35]:
```suggestion
+  <launchable type="desktop-id">io.github.xaos_project.XaoS.desktop</launchable>
```

File: io.github.xaos_project.XaoS.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+          tag-pattern: ^release-([\d.]+)$
+      - type: shell
+        commands:
+          - sed -i 's@#define DATAPATH "/usr/share/XaoS"@#define DATAPATH "/app/share/XaoS"@' src/include/config.h
+      - type: patch
+        path: appdata1.patch
+      - type: patch
+        path: appdata2.patch
+      - type: patch
+        path: appdata3.patch
```
Comment (hfiguiere) [inline io.github.xaos_project.XaoS.yml:41]: not sure why you have 3 patches like that.

File: xaos.appdata.xml (hunk 2 above)
Comment (Mailaender) [reply xaos.appdata.xml:3]: They changed it so many times since last release that it is inconvenient to patch.

File: io.github.xaos_project.XaoS.yml (hunk 4 above)
Comment (Mailaender) [reply io.github.xaos_project.XaoS.yml:41]: To track upstream changes separately to my pull request.

**Comment by Mailaender** [comment]
bot, build io.github.xaos_project.XaoS

File: xaos.appdata.xml (hunk 2 above)
Comment (hfiguiere) [reply xaos.appdata.xml]: also pro-tip: since you get the upstream source from git, just checkout the git tree, change the files in the tree, and then `git diff`. (or commit and then `git show`). Redirect that into the patch file.
